src/test/java/org/junitpioneer/jupiter/CartesianProductTestExtensionTests.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter;
+
+import static org.junitpioneer.testkit.assertion.PioneerAssert.assertThat;
+
+import java.util.concurrent.TimeUnit;
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junitpioneer.testkit.ExecutionResults;
+import org.junitpioneer.testkit.PioneerTestKit;
+
+public class CartesianProductTestExtensionTests {
+
+	@Nested
+	class StandardBehaviouralTests {
+
+		@CartesianProductTest(value = { "0", "1" })
+		@DisplayName("Does nothing if there are no parameters")
+		void empty() {
+		}
+
+		@CartesianProductTest(value = { "0", "1", "2" })
+		@DisplayName("Runs for each parameter once for single parameter")
+		void singleParameter(String param) {
+			int value = Integer.parseInt(param);
+			Assertions.assertThat(value).isBetween(0, 2);
+		}
+
+		@CartesianProductTest({ "0", "1" })
+		void threeBits(String a, String b, String c) {
+			int value = Integer.parseUnsignedInt(a + b + c, 2);
+			Assertions.assertThat(value).isBetween(0b000, 0b111);
+		}
+
+		@CartesianProductTest
+		@DisplayName("S ⨯ T ⨯ U")
+		void nFold(String string, Class<?> type, TimeUnit unit, TestInfo info) {
```

Comment:
What does `nFold` mean?

Reply:
No clue, this is copy-pasted from JUnit examples.

Reply:
What would be a good alternative?

Reply:
As I don't understand it it can't give a good alternative.

Reply:
@Bukama If you think the new introduction properly explains why this example is named `nFold` I will close this discussion.

Reply:
Didn't thought this "discussion" is still relevant. The explanation of `nFold` is in the into and so I'm totally fine :)